lib/ansible/modules/cloud/vmware/vmware_host_dns.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Christian Kotte <christian.kotte@gmx.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_dns
+short_description: Manage DNS configuration of an ESXI host system
+description:
+- This module can be used to configure DNS for ESXi host systems in given vCenter infrastructure.
+- It's only possible to configured DNS for the default TCP/IP stack.
+version_added: 2.8
+author:
+- Christian Kotte (@ckotte)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  host_name:
+    description:
+    - Host name
+    type: str
+    required: True
+  domain:
+    description:
+    - Domain
+    type: str
+    required: True
+    aliases: ['domain_name']
+  dns_servers:
+    description:
+    - DNS servers
+    type: list
+    required: True
+  search_domains:
+    description:
+    - Search domains
+    type: list
+    required: False
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This is required parameter if C(cluster_name) is not specified.
```

Comment:
This parameter is required if C(cluster_name) is not specified.

Reply:
I copied that from one of @Akasurde's modules :)